api/users/permissions.py

```diff
@@ -47,3 +49,63 @@ def has_permission(self, request, view):
     def has_object_permission(self, request, view, obj):
         assert isinstance(obj, OSFUser), f'obj must be a User, got {obj}'
         return not obj.is_registered
+
+
+class UserMessagePermissions(permissions.BasePermission):
+    """
+    Custom permission to allow only institutional admins to create certain types of UserMessages.
+    """
+    def has_permission(self, request, view) -> bool:
+        """
+        Validate if the user has permission to perform the requested action.
+        Args:
+            request: The HTTP request.
+            view: The view handling the request.
+        Returns:
+            bool: True if the user has the required permission, False otherwise.
+        """
+        if request.method != 'POST':
+            return False
```

Comment:
I think this should be handled by other mechanisms, such as the view class. This would cause a 403 for a non-POST, whereas it should get a 405 Method Not Allowed.

Reply:
Good catch because it's only a `generic.CreateView` all other methods are 405